### PR TITLE
Support Kronecker sum

### PR DESCRIPTION
**Pull request by ish-kafel**

### CODE REVIEW

File: src/matrix.js [hunk 1]
```diff
@@ -1119,6 +1119,18 @@ export class AbstractMatrix {
     return result;
   }
 
+  kroneckerSum(other) {
+    other = Matrix.checkMatrix(other);
+    if (!this.isSquare() || !this.isSquare()) {
```
Comment (targos) [inline src/matrix.js:1124]:
```suggestion
    if (!this.isSquare() || !other.isSquare()) {
```

File: src/matrix.js [hunk 2]
```diff
@@ -1119,6 +1119,18 @@ export class AbstractMatrix {
...
+      throw new Error('Kronecker Sum needs two Square Matrices');
+    }
+    let m = this.rows;
+    let n = other.rows;
+    let AxI = this.kroneckerProduct(Matrix.eye(n, n));
+    let IxB = Matrix.eye(m, m).kroneckerProduct(other);
+    return Matrix.add(AxI, IxB);
```
Comment (targos) [inline src/matrix.js:1131]:
```suggestion
    return AxI.add(IxB);
```

I think this should still work, will avoiding to create a third matrix.

Comment (ish-kafel) [reply src/matrix.js:1131]: Thank you for your review!

**Review by targos — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**